### DISCUSSION TITLE
Is Patrol a widget or integration test tool?

### DISCUSSION THREAD

**Discussion by devwela** [leancodepl/patrol#1081] [Q&A] [answered]
Is Patrol a widget or integration test tool?

**Comment by bartekpacia** [comment, marked as answer]
Patrol is both!

It has 2 main features: [custom finders](https://patrol.leancode.co/finders/overview) and [native automation](https://patrol.leancode.co/native/overview).

You can use custom finders in your widget tests and integration tests on all platforms.

Native automation works only on mobile (Android, iOS) and only during integration tests.

**Reply by devwela** [reply]
Thanks!
